toolchain/build-toolchain.sh

```diff
@@ -175,13 +177,60 @@ build_gcc_stage2() {
     log "installing GCC (stage2)"
     make install-gcc install-target-libgcc
 
-    # TODO: build libstdc++ -> for now it fails because of libphoenix missing features
-
     # FIXME: keep the symlink for now until install dir changes in libphoenix and kernel would be well-propagated
     #rm -r "${SYSROOT:?}/usr"
     popd > /dev/null
 }
 
+build_libstdcpp() {
+    # use new compiler for the below builds
+    OLDPATH="$PATH"
+    PATH="$TOOLCHAIN_PREFIX/bin":$PATH
+
+    # set flags for arm to guarantee PIC for libstdc++
+    TARGETFLAGS=
+    if [ "$TARGET" == "arm-phoenix" ]; then
+        export CFLAGS="-fPIC -fPIE"
+        TARGETFLAGS="-fPIC -fPIE"
```

Comment:
Add PIC flags to EXTRA_FLAGS as suggested. Anyway that's not going to work for non-MMU targets which requires:
`EXTRA_FLAGS += "-fpic -fpie -msingle-pic-base -mno-pic-data-is-text-relative"`
So it look like we need to use different PIC flags for MMU and non-MMU targets.